src/devices/machine/ram.cpp

```diff
@@ -212,3 +167,45 @@ uint32_t ram_device::default_size(void) const
 {
 	return parse_string(m_default_size);
 }
+
+
+//-------------------------------------------------
+//  extra_options
+//-------------------------------------------------
+
+const std::vector<uint32_t> &ram_device::extra_options(void) const
+{
+	if (m_extra_options_string && m_extra_options.empty())
+		m_extra_options = calculate_extra_options(m_extra_options_string, nullptr);
+	return m_extra_options;
+}
+
+
+//-------------------------------------------------
+//  calculate_extra_options
+//-------------------------------------------------
+
+std::vector<uint32_t> ram_device::calculate_extra_options(const char *extra_options_string, std::string *bad_option)
+{
+	std::vector<uint32_t> result;
+	std::string options(extra_options_string);
+
+	bool done = false;
+	for (int start = 0, end = options.find_first_of(','); !done; start = end + 1, end = options.find_first_of(',', start))
```

Comment:
you should be using `std::string::size_type` - `std::string::npos` generally can't be represented by an `int`.